src/lib/comms/include/sol-socket.h

```diff
@@ -276,30 +277,32 @@ int sol_socket_set_write_monitor(struct sol_socket *s, bool on);
  * message contents.
  *
  * @param s The value got with @c sol_socket_ip_new()
- * @param buf The data buffer that will be used to receive the data.
- * @param len The size of @a buf
+ * @param buffer the data buffer that will be used to receive the data.
+ * If @a buffer does not has the flags SOL_BUFFER_FLAGS_FIXED_CAPACITY or
+ * SOL_BUFFER_FLAGS_MEMORY_NOT_OWNED this functions will allocate/re-allocate.
+ * The @a buffer must be initialized.
```

Comment:
Maybe these docs that cite buffer flags could have a `@see sol_buffer_flags`? Don't know if it works, though - need to check.